### ISSUE TITLE
The `wrong_self_convention` lint should be applied to traits too

### ISSUE THREAD

**Issue by stanislav-tkach** [rust-lang/rust-clippy#6307]
In the following code the `wrong_self_convention` is triggered for the `S` struct only (the `to_a` method), but not for the trait declaration (the `to_b` method):

```rust
struct S {}

impl S {
    fn to_a(self) {}
}

trait T: Sized {
    fn to_b(self) {}
}
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (18bf6b4 2020-10-07)
- `rustc -Vv`:
  ```
  rustc 1.47.0 (18bf6b4f0 2020-10-07)
  binary: rustc
  commit-hash: 18bf6b4f01a6feaf7259ba7cdae58031af1b7b39
  commit-date: 2020-10-07
  host: x86_64-unknown-linux-gnu
  release: 1.47.0
  LLVM version: 11.0
  ```
